src/core/server/index.ts

```diff
@@ -33,16 +33,14 @@ export class Server {
   private readonly legacy: LegacyCompatModule;
   private readonly log: Logger;
 
-  constructor(
-    private readonly configService: ConfigService,
-    logger: LoggerFactory,
-    private readonly env: Env
-  ) {
+  constructor(configService: ConfigService, logger: LoggerFactory, private readonly env: Env) {
     this.log = logger.get('server');
 
     this.http = new HttpModule(configService.atPath('server', HttpConfig), logger);
-    this.plugins = new PluginsModule(configService, logger, env);
-    this.legacy = new LegacyCompatModule(configService, logger, env);
+
+    const core = { env, configService, logger };
```

Comment:
I don't usually care that much about naming, but I think this one is a little too ambiguous. The notion of "core" is generic enough to be confusing in its own right, and we make that worse by referring to multiple different things core.

Let's have "kibana core" refer to the `src/core` module. If you're talking in the context of the server, "kibana core" might alternatively refer to `src/core/server`, which is the only module that is relevant to you. If you're talking in the context of the browser, then `src/core/public`.

Perhaps something like "base services" would be more appropriate here?

Reply:
Yeah, it's confusing, I didn't use `services` because of `env` that is not a service like `configService`, but maybe we can call it a "service" in a broader sense, so `base services` works for me - will rename.